src/app/filecoin-plus/allocators/utils/getDatacapStats.ts

```diff
@@ -0,0 +1,31 @@
+import { API_URLS } from '@/constants/apiUrls'
+
+import {
+  DatacapSchema,
+  type BaseDatacap,
+  type Datacap,
+} from '../schemas/DatacapSchema'
+
+export async function getDatacapStats(): Promise<Array<Datacap>> {
+  try {
+    const { data } = await fetchDatacapStats()
+    return parseDatacapData(data)
+  } catch (error) {
+    console.error(error)
+    return []
+  }
+}
+
+async function fetchDatacapStats() {
+  const response = await fetch(API_URLS.datacapStats)
+  if (!response.ok) {
+    throw new Error(`Failed to fetch datacap stats, status: ${response.status}`)
+  }
+  return response.json()
+}
+
+function parseDatacapData(data: Array<BaseDatacap>) {
+  return data.map((item: BaseDatacap) => {
+    return DatacapSchema.parse(item)
+  })
+}
```

Comment:
How about something like that? It's a bit simpler.


```suggestion
import { API_URLS } from '@/constants/apiUrls'

import {
  GetDatacapStatsResponseSchema,
  type Datacap,
} from '../schemas/DatacapSchema'

export async function getDatacapStats(): Promise<Array<Datacap>> {
  const response = await fetch(API_URLS.datacapStats)
  const json = await response.json()
  const parsedJson = GetDatacapStatsResponseSchema.parse(json)
  return parsedJson.data
}
```

If you like it, you would have to create this schema:

```
export const GetDatacapStatsResponseSchema = z.object({
  data: z.array(BaseDatacapSchema),
})
```

Reply:
I kind of prefer the modular approach with error handling, but if you think it's clearer, don't mind going with it - @mirhamasala any thoughts?

Reply:
In my opinion, all the error handling we need is in `allocators/components/AllocatorsTableSection.tsx`.

But I couldn't test due to the API rate limits so, maybe this is needed.

Reply:
You might be talking about two things? Charly's approach could still be reworked to be modular? 

```
import { API_URLS } from '@/constants/apiUrls'
import { GetDatacapStatsResponseSchema, type Datacap } from '../schemas/DatacapSchema'

export async function getDatacapStats() {
 const json = await fetchDatacapStats()
 return parseDatacapStats(json)
}

async function fetchDatacapStats() {
 const response = await fetch(API_URLS.datacapStats)
 return response.json()
}

function parseDatacapStats(json: unknown) {
 return GetDatacapStatsResponseSchema.parse(json).data
}
```

Reply:
What do you all think about using some `revalidate`? Something like

```
 const response = await fetch(API_URLS.datacapStats, {
    next: { revalidate: 3600 } // Cache for 1 hour 
  })
```